sync/sync-api/src/main/java/com/duckduckgo/sync/api/SyncStateMonitor.kt

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2023 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.sync.api
+
+import kotlinx.coroutines.flow.Flow
+
+interface SyncStateMonitor {
+
+    /**
+     * Returns a flow of Sync state changes
+     * It follows the following truth table:
+     * * when the user is not signed with Sync is disabled the flow will emit a [SyncState.OFF]
+     * * else it will map the state to the last [SyncAttempt]
+     */
+    fun syncState(): Flow<SyncState>
+
+    /**
+     * Checks if user is signed in on this device.
+     */
+    fun isUserSignedInOnDevice(): Boolean
+}
+
+enum class SyncState {
```

Comment:
This is public API, can you add some docs to know what every state means and more importantly how does it transition from one state to any other

Reply:
yep